src/modules/pmc/src/lib/PmcBase.cpp

```diff
@@ -561,3 +607,130 @@ std::string PmcBase::Trim(const std::string& str, const std::string& trim)
+    {
+        OsConfigLogError(PmcLog::Get(), "Failed to open directory %s, cannot list files", directory);
+    }
+
+    return result;
+}
+
+int PmcBase::ConfigureSources(const std::map<std::string, std::string> sources)
+{
+    int status = MMI_OK;
+
+    for (auto& source : sources)
+    {
+        m_executionState.SetExecutionState(ExecutionState::StateComponent::running, ExecutionState::SubStateComponent::modifyingSources, source.first);
+        std::string sourceFileName = source.first + ".list";
+        std::string sourcesFilePath = m_sourcesConfigurationDirectory + sourceFileName;
+        OsConfigLogInfo(PmcLog::Get(), "Starting to configure source(s) file: %s", sourcesFilePath.c_str());
+
+        // Delete file when provided map value is empty
+        if (source.second.empty())
+        {
+            if (FileExists(sourcesFilePath.c_str()))
+            {
+                status = remove(sourcesFilePath.c_str());
+            }
+            else
+            {
+                OsConfigLogInfo(PmcLog::Get(), "Nothing to delete. Source(s) file: %s does not exist", sourcesFilePath.c_str());
+            }
+
+            if (status != MMI_OK)
+            {
+                OsConfigLogError(PmcLog::Get(), "Failed to delete source(s) file %s with status %d. Stopping configuration for further sources", sourcesFilePath.c_str(), status);
+                m_executionState.SetExecutionState(ExecutionState::StateComponent::failed, ExecutionState::SubStateComponent::modifyingSources, source.first);
+                return errno;
+            }
+        }
+        else
+        {
+            std::ofstream output(sourcesFilePath);
+
+            if (output.fail())
+            {
+                OsConfigLogError(PmcLog::Get(), "Failed to create source(s) file %s. Stopping configuration for further sources", sourcesFilePath.c_str());
+                m_executionState.SetExecutionState(ExecutionState::StateComponent::failed, ExecutionState::SubStateComponent::modifyingSources, source.first);
+                output.close();
+                return errno;
+            }
+
+            output << source.second << std::endl;
+            output.close();
+        }
+    }
+
+    m_executionState.SetExecutionState(ExecutionState::StateComponent::running, ExecutionState::SubStateComponent::updatingPackagesLists);
+    status = RunCommand(g_commandAptUpdate, nullptr);
+
+    if (status != MMI_OK)
+    {
+        OsConfigLogError(PmcLog::Get(), "Refresh package lists failed with status %d", status);
+        status == ETIME ? m_executionState.SetExecutionState(ExecutionState::StateComponent::timedOut, ExecutionState::SubStateComponent::updatingPackagesLists)
+                        : m_executionState.SetExecutionState(ExecutionState::StateComponent::failed, ExecutionState::SubStateComponent::updatingPackagesLists);
+    }
+    else
+    {
+        OsConfigLogInfo(PmcLog::Get(), "Successfully updated package lists");
```

Comment:
> OsConfigLogInfo(PmcLog::Get(), "Successfully updated package lists"); [](http://example.com/codeflow?start=7&length=70)

Don't log this or log only if full logging is enabled

Reply:
I deleted some of the LogInfo lines and the rest of the LogInfo in that file is only used with full logging enabled.

Reply:
Success traces can overflow the log repeating same trace again and again during periodic invocation of MmiGet and MmiSet. 

So in general try to not log anything informal from inside these two functions, MmiGet and MmiSet. 

Even full logging enabled, think if indeed that trace is useful, esp when it will repeat itself many times.

This would be a good occasion to review all informal traces written from inside these 2 MMI calls and their invoked internal helpers.

Reply:
I did some more cleanup and deleted LogInfo messages that weren't really necessary.